napvig:
  rate: 100
  synchronous: true
  stop_on_fail: true
  algorithm: napvig_x
  landscape:
    measure_radius: 0.3
    smooth_radius: 0.05
    precision: 500
    dimensions: 2
    min_distance: 0.15
  napvig:
    core:
      step_ahead_size: 0.2
      gradient_step_size: 0.2
      termination_distance: 0.001
      termination_count: 40
    predictive:
      window_length: 5
      randomized:
        randomize_variance: 1
        max_trials: 5
      x:
        step_gain_saturation_distance: 1
        target_reached_threshold: 0.05
        angle_search:
          min: rad(-pi)
          max: rad(pi)
          step: rad(pi/4)
  debug:
    landscape_test:
      range_min: -0.8
      range_max: 0.8
      range_step: 0.04
      draw: none   # {none, value, grad}
    publish_measures: true
    publish_history: true